### PR TITLE
Review: CDAP 19817 - Footer Panel

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
Feature - Footer Panel UI

## Description

1. This PR contains the UI for the Grid Page's Footer Panel
2. The UI has been developed with static data for all tabs
3. The Meta Info tab has real-time information based on Grid Data

## Related PR
This pr was made because the previous one: https://github.com/cdapio/cdap-ui/pull/738 had too many conflicts in the squash merge. This pr cleans the commit history but is otherwise identical to it.

## Files to be reviewed as part of this PR
Please review all files that are part of this PR

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19817

## Test Plan
FE_37_192 - Verify the footer panel on the grid page is displayed or not - Pass
FE_37_193 - Verify the footer panel on the grid page has the column view panel icon or not. - Pass
FE_37_194 - Verify the column view panel icon by mouse hovering. - Pass
FE_37_195 - Verify the column view panel icon by mouse hovering the tooltip should be displayed. - Pass
FE_37_196 - Verify whether the footer panel has a separate column with "Current data - Rows & Columns" text or not. - Pass
FE_37_197 - Verify whether the footer panel has a separate column with "Zoom in Zoom out with percentage symbol & Number" text or not. - Pass
FE_37_198 - Verify whether the footer panel has a separate column with "Directives" text or not. - Pass
FE_37_199 - Verify whether the footer panel has a separate column with "Recipe Steps with no. count" text or not. - Pass
FE_37_200 - Verify all footer panel columns except the Text Column, by mouse hovering. - Pass

## Screenshots
1. Footer Panel UI
Screenshot 2022-11-07 at 7 24 53 PM

2. Tooltip for Column View Panel Tab Icon in the Footer Panel
Screenshot 2022-11-07 at 7 25 06 PM

### CODE REVIEW

**Review by seanfeld — APPROVED** [review]

**Review by radhikav1 — APPROVED** [review]

File: app/cdap/components/FooterPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  border-left: 1px solid rgba(57, 148, 255, 0.4);
+  cursor: pointer;
+`;
+
+export interface ITableMetaInfoTabProps {
+  rowCount: number;
+  columnCount: number;
+}
+
+const LargeBox = styled(Box)`
```
Comment (mrahanjam) [inline app/cdap/components/FooterPanel/index.tsx]: I highly recommend putting the style related code in a separate file called `styles.tsx` in cases where there's over 100 lines of code for styling like this file.

File: app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx [hunk 2]
```diff
@@ -112,10 +112,11 @@ export default function OngoingDataExploration() {
               pathname: `/ns/${getCurrentNamespace()}/wrangler-grid/${`${item[4].workspaceId}`}`,
               state: {
                 from: T.translate('features.WranglerNewUI.Breadcrumb.labels.wrangleHome'),
-                path: T.translate('features.WranglerNewUI.Breadcrumb.params.wrangeHome'),
+                path: T.translate('features.WranglerNewUI.Breadcrumb.params.wrangleHome'),
```
Comment (mrahanjam) [inline app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx]: `wranglerHome` instead of `wrangleHome`?

**Review by mrahanjam — APPROVED** [review]
LGTM overall, left a couple of comments.

File: app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx (hunk 2 above)
Comment (ManishaGundapuneedi) [reply app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx]: Updated.

File: app/cdap/components/FooterPanel/index.tsx (hunk 1 above)
Comment (ManishaGundapuneedi) [reply app/cdap/components/FooterPanel/index.tsx]: Moved to styles.tsx file

File: app/cdap/components/FooterPanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  return (
+    <TabsWrapper data-testid="footer-panel-wrapper">
+      <CustomTooltip title={`${T.translate(`${PREFIX}.columnViewPanel`)}`}>
+        <ColumnViewBox data-testid="footer-panel-column-view-panel-tab">{ColumnIcon}</ColumnViewBox>
+      </CustomTooltip>
+      <LargeBox data-testid="footer-panel-meta-info-tab">
+        <Label data-testid="footerpanel-simple-label">
+          <>
+            {`${T.translate(`${PREFIX}.currentData`)}
```
Comment (njbriggs) [inline app/cdap/components/FooterPanel/index.tsx]: This should be rewritten so the values `rowCount` and `columnCount` are passed as params, not constructed piecemeal. I think this is fixed in other PRs?

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/FooterPanel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import org.junit.Assert;
+
+public class FooterPanel {
+    @Given("Navigate to Home Page to test footer")
```
Comment (njbriggs) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/FooterPanel.java]: Please indent Java files 2 spaces per tab

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java [hunk 5]
```diff
@@ -24,8 +24,8 @@ private static double getRandomArbitrary(int min, int max) {
   public static final int TEST_TIMEOUT_TIME = 10000;
   public static final int RETRY_INTERVAL = 2000;
   public static final String TEST_TIMEOUT_MESSAGE = "Timed out after" +
-    String.valueOf(TEST_TIMEOUT_TIME / 1000) +
-    "seconds";
+      String.valueOf(TEST_TIMEOUT_TIME / 1000) +
```
Comment (njbriggs) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java]: Please don't change this indentation

**Review by njbriggs — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/FooterPanel.java (hunk 4 above)
Comment (sudhadivami) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/FooterPanel.java]: Done

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java (hunk 5 above)
Comment (sudhadivami) [reply src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java]: sure

File: app/cdap/components/FooterPanel/index.tsx (hunk 3 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/FooterPanel/index.tsx]: Yes, this is fixed in other PRs.
Fixed it here as well.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**